simple_certmanager/forms.py

```diff
@@ -17,3 +21,54 @@ def __init__(self, *args, **kwargs):
     class Meta:
         model = Certificate
         fields = "__all__"
+
+
+class SigningRequestAdminForm(forms.ModelForm):
+    certificate = forms.FileField(
+        required=False,
+        help_text="Upload the public certificate file here. This will be used to verify the signature against the CSR and create the certificate instance.",
+    )
+
+    class Meta:
+        model = SigningRequest
+        fields = "__all__"
+
+    def __init__(self, *args, **kwargs):
+        super(SigningRequestAdminForm, self).__init__(*args, **kwargs)
+        # Check if private key and CSR are empty
+        private_key_empty = not self.instance.private_key
+        csr_empty = not self.instance.csr
+        if private_key_empty or csr_empty:
+            self.fields["certificate"].widget = forms.HiddenInput()
+        else:
+            self.fields["certificate"].widget = forms.FileInput()
+
+    def clean(self):
+        cleaned_data = super().clean()
+        certificate_file = cleaned_data.get("certificate")
+        if certificate_file:
+            try:
+                certificate_data = certificate_file.read()
+                certificate = x509.load_pem_x509_certificate(certificate_data)
+                csr = x509.load_pem_x509_csr(self.instance.csr.encode())
+                # Check if the certificate matches the CSR
+                public_key = csr.public_key()
+                public_key.verify(
+                    certificate.signature,
+                    certificate.tbs_certificate_bytes,
+                    padding.PKCS1v15(),
+                    certificate.signature_hash_algorithm,
+                )
```

Comment:
@CharString does this look right?

This padding argument seems... arbitrary?

Reply:
This may not be correct. The same padding scheme should be used during verification as the issuer used during signing. That can vary from issuer to issuer. If asking the user is unfeasible (the library is called "simple" after all :)) we might be able to do heuristics on the cipher. Depending on if PKCS 1.5 or the newer version is used, the cipher may look recognisably different.